go-controller/pkg/libovsdbops/chassis.go

```diff
@@ -115,3 +140,142 @@ func DeleteChassisWithPredicate(sbClient libovsdbclient.Client, p chassisPredica
 	err := m.Delete(opModels...)
 	return err
 }
+
+// CreateOrUpdateRemoteChassis creates a chassis record if not present. It also
+// creates the Encap record required when creating a chassis record.
+// If the chassis record already exists, it marks the chassis as a remote
+// by setting the options:is-remote=true.
+// Note: Normally ovn-controller creates a chassis record when it connects to
+// OVN Southbound db.  But with multiple zones deployment, ovn-controller will create the
+// chassis record in its own local zone Southbound db. ovn-k network controller
+// manager needs to create a chassis records for these remote nodes in its zone Southbound db.
+func CreateOrUpdateRemoteChassis(sbClient libovsdbclient.Client, hostname, name, ip string) error {
+	ch := &sbdb.Chassis{
+		Hostname: hostname,
+		Name:     name,
+	}
+
+	chassis, err := GetChassis(sbClient, ch)
+	if err != nil {
+		if !errors.Is(err, client.ErrNotFound) {
+			return err
+		}
+
+		// Remote chassis record not found. Create it.
+		chassis = &sbdb.Chassis{
+			Hostname: hostname,
+			Name:     name,
+		}
+	}
+
+	if chassis.OtherConfig != nil {
+		if strings.ToLower(chassis.OtherConfig["is-remote"]) == "true" {
+			// Nothing to do. The chassis is already marked as remote.
+			return nil
+		}
+		chassis.OtherConfig["is-remote"] = "true"
+	} else {
+		chassis.OtherConfig = map[string]string{
+			"is-remote": "true",
+		}
+	}
+
+	encap := &sbdb.Encap{
+		Type:        "geneve",
+		ChassisName: name,
+		IP:          ip,
+		Options:     map[string]string{"csum": "true"},
+	}
+	m := newModelClient(sbClient)
+	opModels := []operationModel{
+		{
+			Model:          encap,
+			ModelPredicate: func(item *sbdb.Encap) bool { return item.Type == encap.Type && item.IP == encap.IP },
+			OnModelUpdates: []interface{}{
+				&encap.Options,
+			},
```

Comment:
Should the Encap ChassisName be updated?

Interesting that Options is updated and not mutated (mutation would make the method more reusable for other purposes).